src/zenml/client_lazy_loader.py

```diff
@@ -208,6 +208,9 @@ def _inner(*args: Any, **kwargs: Any) -> Any:
                     with contextlib.suppress(ValueError):
                         kwargs[k] = ClientLazyLoader(**v).evaluate()
 
+                # Why do we check for `isinstance(v, ClientLazyLoader)` for the
```

Comment:
This works as expected and evaluates at runtime because `intermediate` it passed as an arg:

```python
@pipeline(enable_cache=False, model=Model(name=MODEL_NAME))
def test_pipeline():
    intermediate = Client().get_model(MODEL_NAME).latest_version_id

    cll = (
        Client()
        .get_model_version(
            MODEL_NAME, intermediate
        )
        .get_artifact(ARTIFACT_NAME)
    )
```

If we pass it as a keyword argument instead, the `intermediate` lazy loader already gets evaluated at pipeline compilation time
```python
@pipeline(enable_cache=False, model=Model(name=MODEL_NAME))
def test_pipeline():
    intermediate = Client().get_model(MODEL_NAME).latest_version_id

    cll = (
        Client()
        .get_model_version(
            MODEL_NAME, model_version_name_or_number_or_id=intermediate
        )
        .get_artifact(ARTIFACT_NAME)
    )
```

Reply:
@avishniakov I tried to understand the reason for the `evaluate_all_lazy_load_args_in_client_methods` decorator, and the only thing I could come up with were these nested lazy loaders. Then I ran into this issue with the args/kwargs and was wondering what the expected behaviour would be in your opinion